src/Sparrow/Utils/NativeMemoryCleaner.cs

```diff
@@ -104,18 +129,14 @@ public void CleanNativeMemory(object state)
                 if (Logger.IsErrorEnabled)
                     Logger.Error("Error during cleanup.", e);
             }
-            finally
-            {
-                if (lockTaken)
-                    Monitor.Exit(_lock);
-            }
         }
 
         public void Dispose()
         {
 #if !NETSTANDARD1_3
             using (var waitHandle = new ManualResetEvent(false))
             {
+                _disposed = true;
```

Comment:
It's interesting that it was missing. Can you say something more about that? How did you stumble upon this?

Reply:
Just by looking at the code... total surprise on my side on how we missed that one.